privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardHybridViewModel.kt

```diff
@@ -157,6 +161,36 @@ class PrivacyDashboardHybridViewModel @Inject constructor(
         val cosmetic: Boolean? = false,
     )
 
+    data class RemoteFeatureSettingsViewState(
+        val primaryScreen: PrimaryScreenSettings = PrimaryScreenSettings(),
+    ) {
+        companion object {
+            fun fromSettings(settings: PrivacyDashboardRemoteFeature): RemoteFeatureSettingsViewState {
+                val layout = if (settings.protectionToggleHighlight().isEnabled()) {
+                    LayoutType.HIGHLIGHTED_PROTECTIONS_TOGGLE.value
+                } else {
+                    LayoutType.DEFAULT.value
+                }
+
+                return RemoteFeatureSettingsViewState(
+                    primaryScreen = PrimaryScreenSettings(
+                        layout = layout,
+                    ),
+                )
+            }
+        }
+    }
```

Comment:
I would remove the `fromSettings()` (see also more changes below)
```suggestion
data class RemoteFeatureSettingsViewState(val primaryScreen: PrimaryScreenSettings = PrimaryScreenSettings())
```

The reason is that `isEnabled()` checks on a feature flag may require IO, so we should do them off the main thread.